src/platform/Darwin/PlatformManagerImpl.cpp

```diff
@@ -147,7 +147,7 @@ CHIP_ERROR PlatformManagerImpl::_PostEvent(const ChipDeviceEvent * event)
 bool PlatformManagerImpl::_IsChipStackLockedByCurrentThread() const
 {
     // Assume our caller knows what they are doing in terms of concurrency if the work queue is suspended.
-    return IsWorkQueueCurrentQueue() || mWorkQueueState.load() == WorkQueueState::kSuspended;
+    return IsWorkQueueCurrentQueue() || mWorkQueueState.load() == WorkQueueState::kSuspended || (mChipStackIsLocked && (pthread_equal(pthread_self(), mChipStackLockOwnerThread)));
```

Comment:
Unfortunately, this is completely broken.

The way stack locking in Matter works is:

1. There is a lock that you can lock/unlock.
2. The event loop takes the lock while processing an event.  This ensures that event processing cannot thread-race against someone holding the lock.

For Darwin, item 2 is just not true: there is no custom event loop that takes a lock and processes events, because events are just managed via a dispatch queue.